aql/src/main/java/org/ehrbase/aql/dto/path/predicate/PredicateHelper.java

```diff
@@ -0,0 +1,388 @@
+/*
+ * Copyright (c) 2022 vitasystems GmbH and Hannover Medical School.
+ *
+ * This file is part of project openEHR_SDK
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehrbase.aql.dto.path.predicate;
+
+import static org.ehrbase.aql.parser.AqlToDtoVisitor.buildLogicalOperator;
+
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.ehrbase.aql.dto.condition.ConditionComparisonOperatorSymbol;
+import org.ehrbase.aql.dto.condition.LogicalOperatorDto;
+import org.ehrbase.aql.dto.condition.ParameterValue;
+import org.ehrbase.aql.dto.condition.SimpleValue;
+import org.ehrbase.aql.dto.condition.Value;
+import org.ehrbase.aql.dto.path.AqlPath;
+import org.ehrbase.util.exception.SdkException;
+
+/**
+ * @author Stefan Spiska
+ */
+public class PredicateHelper {
+
+    public static final String[] PREDICATE_DIVIDERS = {" and ", " AND ", " or ", " OR ", ","};
```

Comment:
Exposing a final array may be problematic. It is sorted (over and over again) in AqlPath::split.
Maybe somehow wrap the search operation here?
Also, this seems to introduce a circular dependency with AqlPath.